pkg/apis/controlplane/types.go

```diff
@@ -560,6 +560,9 @@ type BundleFileServer struct {
 	// The URL of the bundle file server. It is set with format: scheme://host[:port][/path],
 	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
 	URL string
+	// HostPublicKey specifies the only host public key that will be accepted when connecting to
+	// the file server. If omitted, any host key will be accepted, which is not recommended.
+	HostPublicKey []byte
```

Comment:
I don't find this field being set. The conversion function should also be updated.

Reply:
That code only sets the "internal" type, which will be converted into the "unversioned" API type in https://github.com/antrea-io/antrea/blob/53d253af9e19d9025c0232814666ea52103f2a80/pkg/controller/supportbundlecollection/store/collection.go#L99-L101
The missing part is in the conversion between the internal type and the unversioned API type.

Reply:
Thanks, I added the missing code. I think I asked in the past, but I don't remember why we need 2 separate types here (unversioned API type and internal storage type)?

Reply:
When we implemented NetworkPolicy, we introduced the internal storage type to address the performance issue that it can't efficiently figure out whether a resource should be sent to a Node, and the pod members that should be sent to a Node, see https://github.com/antrea-io/antrea/blob/129897d6a58c302b02525528fd8943a838803ec1/pkg/controller/types/networkpolicy.go#L41

I guess SupportBundleCollection followed it for determining whether a resource should be sent to a Node. The unversioned type doesn't have a field to indicate that.